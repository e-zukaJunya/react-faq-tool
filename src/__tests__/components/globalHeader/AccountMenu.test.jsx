import AccountMenu from "components/globalHeader/AccountMenu";
import { mount, shallow } from "enzyme";
import React from "react";
import ReactDOM from "react-dom";

// jsdomにcreateRangeを実装
document.createRange = () => ({
  setStart: () => {},
  setEnd: () => {},
  commonAncestorContainer: {
    nodeName: "BODY",
    ownerDocument: document
  }
});

describe("AlertDialog", () => {
  beforeEach(() => {});

  test("mount", () => {
    const wrapper = mount(<AccountMenu />);
    // const wrapper = shallow(<AccountMenu />);
    const dom = wrapper.getDOMNode();
    expect(dom).toMatchSnapshot();
  });

  test("open", () => {
    const wrapper = mount(<AccountMenu />);
    const btn = wrapper.find("button");
    btn.simulate("click");

    const dom = wrapper.getDOMNode();
    expect(dom).toMatchSnapshot();
  });

  // todo ClickAwayListenerのテストができない
  test.skip("close", async () => {
    // const wrapper = mount(<AccountMenu />);
    const wrapper = mount(
      <div>
        <AccountMenu />
        <div id={"otherElm"}></div>
      </div>
    );

    const btn = wrapper.find("button");
    btn.simulate("click");

    const dom = wrapper.getDOMNode();
    expect(dom).toMatchSnapshot();
  });

  test.skip("click logout", () => {
    const wrapper = mount(<AccountMenu />);
    const btn = wrapper.find("li");
    btn.simulate("click");
  });
});
