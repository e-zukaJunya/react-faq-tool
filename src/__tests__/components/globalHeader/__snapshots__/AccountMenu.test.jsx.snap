// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog close 1`] = `
<div>
  <div>
    <button
      aria-controls="primary-search-account-menu"
      aria-haspopup="true"
      aria-label="account of current user"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      style="position: absolute;"
      x-placement="bottom"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="opacity: 1; transform: scale(1, 1); transform-origin: center bottom; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <ul
          class="MuiList-root MuiList-padding"
          id="menu-list-grow"
          role="menu"
          tabindex="-1"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="0"
          >
            ログアウト
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    id="otherElm"
  />
</div>
`;

exports[`AlertDialog mount 1`] = `
<div>
  <button
    aria-controls="primary-search-account-menu"
    aria-haspopup="true"
    aria-label="account of current user"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`AlertDialog open 1`] = `
<div>
  <button
    aria-controls="primary-search-account-menu"
    aria-haspopup="true"
    aria-label="account of current user"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    style="position: absolute;"
    x-placement="bottom"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="opacity: 1; transform: scale(1, 1); transform-origin: center bottom; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <ul
        class="MuiList-root MuiList-padding"
        id="menu-list-grow"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="0"
        >
          ログアウト
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
